frontend/src/components/NewPassword/NewPassword.js

```diff
@@ -22,17 +24,51 @@ class NewPassword extends Component {
     this.setState({ [event.target.name]: event.target.value });
   };
 
+  handleSubmit = (event) => {
+    event.preventDefault();
+    const { token, history } = this.props;
+    const { password, confirmPassword } = this.state;
+    if (password !== confirmPassword) {
+      this.setState({ error: "Passwords don't match" });
+    } else {
+      resetPassword(token, password)
+        .then(() => {
+          history.push('/login');
+        })
+        .catch((error) => {
+          if (error.response.data.errors.reset_password_token[0]) {
+            this.setState({
+              error: `Reset password token ${error.response.data.errors.reset_password_token[0]}.`,
```

Comment:
What does this actually end up returning?

Reply:
I think we'll also want to handle the case where something goes wrong that isn't token-related. Like if the reset password token thing is there, then yeah, display this error. Else display something like, "Something went wrong with our system. Please try again."